enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,362 @@
+payload produced for one of the included architectures.
+
+In the case of `oc adm release new`, the manifestlist flow would be triggered whenever the CVO image in a source image/imagestream is a manifestlist.
+- If the source CVO image is a standard manifest, the generated release payload will also be a manifest.
+- If the source CVO image is a manifestlist, the generated release payload would be a manifestlist(containing a manifest for each architecture
+possessed by the CVO manifestlist).
+
+In either case, oc adm release new would permit non-CVO component images to be manifest or manifestlists and pass them through directly to the
+resultant release manifest(s). If a manifestlist release payload is generated, each architecture specific release payload manifest will reference
+the same pullspecs provided in the input imagestream. Since the internal Openshift registry does not support manifestlists, the release controller
+would use quay.io.
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for heterogeneous
+payloads which CVO will need to look at to make upgrade recommendations. There are a few upgrade scenarios to consider with the addition of
+heterogeneous payloads:
+
+- heterogeneous -> heterogeneous payload upgrade
+- homogeneous -> heterogeneous payload upgrade
+- heterogeneous -> homogeneous payload upgrade
+
+The support for homogeneous payloads will exist so customers running disconnected homogeneous clusters do not incur the cost of mirroring the
+additional architectures if they were to use heterogeneous payloads. This means that in order for CVO to find the current release in the graph, it
+will have to know about the existing homogeneous graph as well in addition to the new heterogeneous graph. There are few options on the best way
+to make update recommendations especially in the case of upgrading from a homogeneous to a heterogeneous payload:
+
+- Have the heterogeneous graph include the homogeneous releases so CVO can make recommendations based on the current homogeneous release image.
+This is complicated by the fact that there will be multiple releases with the same version but different payloads. The trigger to update to a
+heterogeneous payload would be an option provided by the administrator through either the ClusterVersion or Infrastructure API to "move" to the
+heterogeneous graph.
+- Have both the graphs be completely independent of each other and have an option in one of the aforementioned APIs to make the cluster heterogeneous
+and CVO would request the heterogeneous graph, find the current release version and update to it.
+- Have an independent channel for heterogeneous payloads and have administrators switch channels if they want to move between homogeneous and
```

Comment:
This is still an open question.

Reply:
yes so i have listed  the options here. in the "Open questions" section i have pointed back to this section as I have summarized the upgrade strategy overall.

Reply:
What do you see as the risk that is being avoided?  Going from 4.11.0-x86 to 4.11.1-multi should be equivalent to going from 4.11.0-x86 to 4.11.1-x86, with the only difference being that the payload pullspecs will be manifestlists instead of manifests.

The cluster itself would only have x86 nodes regardless, so the actual image content the cluster consumes will be the same in the end.

Reply:
I don't think the risk is high, I just don't want to be troubleshooting two concurrent changes in an uncommonly exercised migration. If we want to phase this in, offer it as a distinct migration step in one minor, then automate it in a later minor after having proven success in the field then I'm fine with that.

Reply:
@sdodson @LalatenduMohanty @wking have we reached  a consensus on the approach we will be taking here?